### PR TITLE
empty-bin command options (recursive and liquid)

### PR DESCRIPTION
**Pull request by foxxelias**
Added the  `-r | --recursive` option to allow containers to be emptied on the floor recursively.

Added the `-l | --liquid` option  and changed the basic behavior of the `empty-bin` command, to move items with the types `LIQUID_MISC` and `DRINK` to the floor you must `explicitly` specify option `-l | --liquid`

Example usage:
Select stockpile with bin, with filled waterskin inside and use `empty-bin -l -r` 
This will recursively empty all containers on the stockpile, draining them of liquid if there is any in them.
<details>
<summary>*pictures*</summary>

image

image2

</details>

### CODE REVIEW

File: empty-bin.lua [hunk 1]
```diff
@@ -3,28 +3,78 @@
...
+        liquids = false
+    },
+    {...}
 
 local function emptyContainer(container)
     local items = dfhack.items.getContainedItems(container)
-
     if #items > 0 then
-        print('Emptying ' .. dfhack.items.getReadableDescription(container))
+        print("Emptying " .. dfhack.items.getReadableDescription(container))
```
Comment (chdoc) [inline empty-bin.lua:18]: Most of our code base uses single quotes for strings.

File: docs/empty-bin.rst [hunk 2]
```diff
@@ -22,4 +22,12 @@ Usage
...
-    empty-bin
+    empty-bin [<options>]
+
+Options
+--------------
+
+``-r|--recursive``
+    Whether to apply the command recursively.
+``-l|--liquids``
+    Include liquids (DRINK and LIQUID_MISC) in the items list to be moved to the ground.
```
Comment (chdoc) [inline docs/empty-bin.rst:33]:
```suggestion
``-r``, ``--recursive``
    Recursively empty containers.
``-l``, ``--liquids``
    Move contained liquids (DRINK and LIQUID_MISC) to the floor, making them unusable.
```

You should also remove the "Note that if there are liquids..." from above.

File: empty-bin.lua [hunk 3]
```diff
@@ -3,28 +3,78 @@
...
+argparse.processArgsGetopt(
+    args,
+    {
+        {
+            "h",
+            "help",
+            handler = function()
+                options.help = true
+            end
+        },
```
Comment (chdoc) [inline empty-bin.lua:50]:
```suggestion
        {  'h', 'help',  handler = function() options.help = true end },
```
Is is customary to put short option handlers onto one line.

**Comment by myk002** [comment]
apologies for updating the branch with a master merge -- I clicked the wrong button.

**Comment by myk002** [comment]
on another note, please feel free to add your name to https://github.com/DFHack/dfhack/blob/develop/docs/about/Authors.rst

**Review by myk002 — APPROVED** [review]

**Pull request merged**